SIGSEGV with mildly invalid template

<!--
Note that this is the issue tracker for OpenShift Origin.

OpenShift Installer issues should be reported at https://github.com/openshift/openshift-ansible/issues
Management Console issues are collected at https://github.com/openshift/origin-web-console/issues
Documentation issues are better reported at https://github.com/openshift/openshift-docs/issues
-->

By playing around, if found a null deref for a rather small (yet invalid) file. It's local and not openshift itself failing. As I was unable to find an easy way to test for failure (like droping the invalid file into a test directory), I didn't provided a test case; it might be that's also only triggering for me, but, as it was already present in 3.7.0, I guess it's not just me.

##### Version
```
oc v3.10.0-alpha.0+43de4df-161
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO
```

##### Steps To Reproduce
1. login into an openshift instance
2. create a template file, let's name it `fail`, with content
```
apiVersion: v1
kind: Template
metadata:
  name: random
objects:   
  - apiVersion: v1
    kind: DeploymentConfig
```
3. `oc new-app -f fail`

##### Current Result
```
--> Deploying template "xxx/random" for "fail" to project xxx

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x22d747d]

goroutine 1 [running]:
github.com/openshift/origin/pkg/util.addDeploymentConfigNestedLabels(0xc4206b1400, 0xc421d0e4b0, 0x2, 0x0, 0xc421d0e4e0)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/pkg/util/labels.go:181 +0x2d
github.com/openshift/origin/pkg/util.AddObjectLabelsWithFlags(0x7b2f500, 0xc4206b1400, 0xc421d0e4b0, 0x2, 0x416218, 0x10)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/pkg/util/labels.go:44 +0x80c
github.com/openshift/origin/pkg/oc/cli/cmd.hasLabel(0xc421d0e4b0, 0xc421354540, 0x4cd616a, 0x3, 0xc420e8a2c8)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/pkg/oc/cli/cmd/newapp.go:676 +0x9c
github.com/openshift/origin/pkg/oc/cli/cmd.(*NewAppOptions).RunNewApp(0xc421356188, 0x0, 0x2)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/pkg/oc/cli/cmd/newapp.go:316 +0x1dcf
github.com/openshift/origin/pkg/oc/cli/cmd.NewCmdNewApplication.func1(0xc420443d40, 0xc4213add70, 0x0, 0x3)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/pkg/oc/cli/cmd/newapp.go:223 +0x10a
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).execute(0xc420443d40, 0xc4213adce0, 0x3, 0x3, 0xc420443d40, 0xc4213adce0)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:603 +0x234
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42042afc0, 0x2061c91, 0xc42042afc0, 0xc420289b90)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:689 +0x2fe
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42042afc0, 0x2, 0xc42042afc0)
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
        /var/tmp/paludis/app-admin-openshift-client-tools-9999/work/openshift-client-tools-9999/_output/local/go/src/github.com/openshift/origin/cmd/oc/oc.go:41 +0x293
```

##### Expected Result
```
--> Deploying template "xxx/random" for "fail" to project xxx

--> Creating resources ...
    error: DeploymentConfig "" is invalid: [metadata.name: Required value: name or generateName is required, spec.template: Required value, spec.selector: Invalid value: map[string]string(nil): selector cannot be empty]
--> Failed
```

--- a/pkg/util/labels.go
+++ b/pkg/util/labels.go
@@ -178,12 +178,18 @@ func AddObjectAnnotations(obj runtime.Object, annotations map[string]string) err
 
 // addDeploymentConfigNestedLabels adds new label(s) to a nested labels of a single DeploymentConfig object
 func addDeploymentConfigNestedLabels(obj *appsapi.DeploymentConfig, labels labels.Set, flags int) error {
+	if obj.Spec.Template == nil {
+		return nil
+	}
+
 	if obj.Spec.Template.Labels == nil {
 		obj.Spec.Template.Labels = make(map[string]string)
 	}
+
 	if err := MergeInto(obj.Spec.Template.Labels, labels, flags); err != nil {
 		return fmt.Errorf("unable to add labels to Template.DeploymentConfig.Template.ControllerTemplate.Template: %v", err)
 	}
+
 	return nil
 }
 

